# Train.csv

---

Nuestras carpetas están organizadas en un principio en "../data/raw/" y tenemos las carpetas:

 `test_images`, `train_images`,
 
 y los archivos:
 
  `sample_submission.csv`, `test_series_descriptions.csv`, `train_label_coordinates.csv`, `train_series_descriptions.csv` y `train.csv`.

---

### Step 1: Problem statement and data collection

**Analysis Objective**

**Context os the analysis**


In [1]:
# Import the dataset to start working with it

import pandas as pd 

df_train = pd.read_csv("../data/raw/train.csv")
df_train.head(3)

study_id spinal_canal_stenosis_l1_l2 spinal_canal_stenosis_l2_l3  \
0   4003253                 Normal/Mild                 Normal/Mild   
1   4646740                 Normal/Mild                 Normal/Mild   
2   7143189                 Normal/Mild                 Normal/Mild   

  spinal_canal_stenosis_l3_l4 spinal_canal_stenosis_l4_l5  \
0                 Normal/Mild                 Normal/Mild   
1                    Moderate                      Severe   
2                 Normal/Mild                 Normal/Mild   

  spinal_canal_stenosis_l5_s1 left_neural_foraminal_narrowing_l1_l2  \
0                 Normal/Mild                           Normal/Mild   
1                 Normal/Mild                           Normal/Mild   
2                 Normal/Mild                           Normal/Mild   

  left_neural_foraminal_narrowing_l2_l3 left_neural_foraminal_narrowing_l3_l4  \
0                           Normal/Mild                           Normal/Mild   
1                           Normal/Mild                           Normal/Mild   
2                           Normal/Mild                           Normal/Mild   

  left_neural_foraminal_narrowing_l4_l5  ... left_subarticular_stenosis_l1_l2  \
0                              Moderate  ...                      Normal/Mild   
1                              Moderate  ...                      Normal/Mild   
2                           Normal/Mild  ...                      Normal/Mild   

  left_subarticular_stenosis_l2_l3 left_subarticular_stenosis_l3_l4  \
0                      Normal/Mild                      Normal/Mild   
1                      Normal/Mild                      Normal/Mild   
2                      Normal/Mild                      Normal/Mild   

  left_subarticular_stenosis_l4_l5 left_subarticular_stenosis_l5_s1  \
0                         Moderate                      Normal/Mild   
1                           Severe                      Normal/Mild   
2                      Normal/Mild                      Normal/Mild   

  right_subarticular_stenosis_l1_l2 right_subarticular_stenosis_l2_l3  \
0                       Normal/Mild                       Normal/Mild   
1                       Normal/Mild                          Moderate   
2                       Normal/Mild                       Normal/Mild   

  right_subarticular_stenosis_l3_l4 right_subarticular_stenosis_l4_l5  \
0                       Normal/Mild                       Normal/Mild   
1                          Moderate                          Moderate   
2                       Normal/Mild                       Normal/Mild   

  right_subarticular_stenosis_l5_s1  
0                       Normal/Mild  
1                       Normal/Mild  
2                       Normal/Mild  

[3 rows x 26 columns]

#### **Files**

**train.csv**

>NOTE:  Some entries have incomplete labels. !!!!!!!!!!!

- `study_id` --> 

- **spinal_canal_stenosis**

    - `spinal_canal_stenosis_l1_l2` -->

    - `spinal_canal_stenosis_l2_l3` -->

    - `spinal_canal_stenosis_l3_l4` -->

    - `spinal_canal_stenosis_l4_l5` -->

    - `spinal_canal_stenosis_l5_s1` -->

- **left_neural_foraminal_narrowing**

    - `left_neural_foraminal_narrowing_l1_l2` -->

    - `left_neural_foraminal_narrowing_l2_l3` -->

    - `left_neural_foraminal_narrowing_l3_l4` -->

    - `left_neural_foraminal_narrowing_l4_l5` -->

    - `left_neural_foraminal_narrowing_l5_s1` -->

- **right_neural_foraminal_narrowing**

    - `right_neural_foraminal_narrowing_l1_l2` -->

    - `right_neural_foraminal_narrowing_l2_l3` -->

    - `right_neural_foraminal_narrowing_l3_l4` -->

    - `right_neural_foraminal_narrowing_l4_l5` -->

    - `right_neural_foraminal_narrowing_l5_s1` -->

- **left_subarticular_stenosis**

    - `left_subarticular_stenosis_l1_l2` -->

    - `left_subarticular_stenosis_l2_l3` -->

    - `left_subarticular_stenosis_l3_l4` -->

    - `left_subarticular_stenosis_l4_l5` -->

    - `left_subarticular_stenosis_l5_s1` -->

- **right_subarticular_stenosis**

    - `right_subarticular_stenosis_l1_l2` -->

    - `right_subarticular_stenosis_l2_l3` -->

    - `right_subarticular_stenosis_l3_l4` -->

    - `right_subarticular_stenosis_l4_l5` -->

    - `right_subarticular_stenosis_l5_s1` -->

>NOTE: Cada etiqueta puede tener diferentes 'scores', estos pueden ser: Normal/Mid, Moderate y Severe.


---

### Step 2: Exploration and data cleaning

- #### 2.1 Summarize DF

In [2]:
# Obtain dimensions

rows, columns = df_train.shape

print(f"The dimensions of this dataset are: {rows} Rows and {columns} Columns")

The dimensions of this dataset are: 1975 Rows and 26 Columns


In [3]:
# Obtain information about data types and non-null values

df_train.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1975 entries, 0 to 1974
Data columns (total 26 columns):
 #   Column                                  Non-Null Count  Dtype 
---  ------                                  --------------  ----- 
 0   study_id                                1975 non-null   int64 
 1   spinal_canal_stenosis_l1_l2             1974 non-null   object
 2   spinal_canal_stenosis_l2_l3             1974 non-null   object
 3   spinal_canal_stenosis_l3_l4             1974 non-null   object
 4   spinal_canal_stenosis_l4_l5             1974 non-null   object
 5   spinal_canal_stenosis_l5_s1             1974 non-null   object
 6   left_neural_foraminal_narrowing_l1_l2   1973 non-null   object
 7   left_neural_foraminal_narrowing_l2_l3   1973 non-null   object
 8   left_neural_foraminal_narrowing_l3_l4   1973 non-null   object
 9   left_neural_foraminal_narrowing_l4_l5   1973 non-null   object
 10  left_neural_foraminal_narrowing_l5_s1   1973 non-null   object
 11  righ

In [4]:
# Check null values

null_var = df_train.isnull().sum().loc[lambda x: x > 0]

num_of_null_var = len(null_var)

print(f"{null_var}\n\nThe number of null variables are {num_of_null_var}")

spinal_canal_stenosis_l1_l2                 1
spinal_canal_stenosis_l2_l3                 1
spinal_canal_stenosis_l3_l4                 1
spinal_canal_stenosis_l4_l5                 1
spinal_canal_stenosis_l5_s1                 1
left_neural_foraminal_narrowing_l1_l2       2
left_neural_foraminal_narrowing_l2_l3       2
left_neural_foraminal_narrowing_l3_l4       2
left_neural_foraminal_narrowing_l4_l5       2
left_neural_foraminal_narrowing_l5_s1       2
right_neural_foraminal_narrowing_l1_l2      8
right_neural_foraminal_narrowing_l2_l3      8
right_neural_foraminal_narrowing_l3_l4      8
right_neural_foraminal_narrowing_l4_l5      8
right_neural_foraminal_narrowing_l5_s1      8
left_subarticular_stenosis_l1_l2          164
left_subarticular_stenosis_l2_l3           82
left_subarticular_stenosis_l3_l4            3
left_subarticular_stenosis_l4_l5            3
left_subarticular_stenosis_l5_s1           11
right_subarticular_stenosis_l1_l2         161
right_subarticular_stenosis_l2_l3 

---------**Conclusiones**---------

- There are a total of 1975 rows (Severity Scores) y 26 columns (enfermedades en discos intervertebrales). AQUÍ PONER EL TARGET QUE BUSCAMOS

- Todas las variables tienen al menos un valor nulo menos `study_id`. `left_subarticular_stenosis_l1_l2` con 164 nulos y `right_subarticular_stenosis_l1_l2` con 161 nulos son las variables con mayor número de valores perdidos.

- The data has 25 categorical characteristics and 1 numerical (`study_id`).

---

- #### 2.2 Eliminate duplicates

In [5]:
# Obtain the number of duplicate values in our dataset

print(f"The number of duplicate values in the dataset is {df_train.duplicated().sum()}")

The number of duplicate values in the dataset is 0
